### DISCUSSION TITLE
org.oscim.layers.vector.geometries package not available?

### DISCUSSION THREAD

**Discussion by DimitrisMantas** [mapsforge/vtm#930] [Q&A] [answered]
Hi, I'm building an Android app using VTM and I my `build.gradle` file looks like this:

```  java
implementation 'org.mapsforge:vtm-android:0.17.0'
implementation 'org.mapsforge:vtm-android:0.17.0:natives-armeabi-v7a'
implementation 'org.mapsforge:vtm-android:0.17.0:natives-arm64-v8a'
implementation 'org.mapsforge:vtm-android:0.17.0:natives-x86'
implementation 'org.mapsforge:vtm-android:0.17.0:natives-x86_64'
implementation 'org.mapsforge:vtm-gdx:0.17.0'
implementation 'org.mapsforge:vtm-android-gdx:0.17.0'
implementation 'org.mapsforge:vtm-android-gdx:0.17.0:natives-armeabi-v7a'
implementation 'org.mapsforge:vtm-android-gdx:0.17.0:natives-arm64-v8a'
implementation 'org.mapsforge:vtm-android-gdx:0.17.0:natives-x86'
implementation 'org.mapsforge:vtm-android-gdx:0.17.0:natives-x86_64'
implementation 'com.badlogicgames.gdx:gdx:1.9.10'
implementation 'com.badlogicgames.gdx:gdx-backend-android:1.9.10'
implementation 'com.caverock:androidsvg:1.4'
```

I'm trying to draw dashed lines on the map, but Android Studio cannot find the `org.oscim.layers.vector.geometries` package, so I can build an appropriate style.

I have also tried to work with `LineStyle` but that does not appear to be working.

Thanks.

**Comment by devemux86** [comment, marked as answer]
See the [integration guide](https://github.com/mapsforge/vtm/blob/master/docs/Integration.md).

For Android you need the sections:
- [Core](https://github.com/mapsforge/vtm/blob/master/docs/Integration.md#core)
- [Android](https://github.com/mapsforge/vtm/blob/master/docs/Integration.md#android)

For advanced overlays you need:
- [JTS geometries](https://github.com/mapsforge/vtm/blob/master/docs/Integration.md#jts-geometries)

**Comment by DimitrisMantas** [comment]
Thanks, its working now. Could you please explain what the `stipple(int)` and `stippleWidth(float)` methods do? They do not seem to documented...

**Comment by devemux86** [comment]
See the [render theme schema](https://github.com/mapsforge/vtm/blob/master/resources/rendertheme.xsd#L203-L208), it contains style documentation.

**Reply by DimitrisMantas** [reply]
All clear now, thanks for your help.
